node/src/requests/debug.rs

```diff
@@ -88,9 +88,13 @@ impl<RequestType: Request, ChainRespondArgsType: ChainRespondArgs> Debug
     for CompletedRequest<RequestType, ChainRespondArgsType>
 {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let (leader, attempts) = {
+            let progress = self.progress.lock().unwrap();
```

Comment:
Unwrap is okay here since this is a mutex. I can barely think if any other scenario where I'm okay with unwrapping errors in production code :sweat_smile: